### ISSUE TITLE
Feature: choose to generate DTOs with Lombok annotations or in the Kotlin language

### ISSUE THREAD

**Issue by massimosiani** [jhipster/generator-jhipster#6296]
##### **Overview of the issue**

DTOs are generated with a standard list of getters and setters. The annotations provided by the Lombok project reduce the boilerplate source code, and they introduce no overhead at runtime.
There is a good plugin for IntelliJ, and I guess other IDEs have nice plugins too.

Going further, a good alternative to the lombok project is the Kotlin Programming Language.

##### **Motivation for or Use Case**

The main motivation for this ticket is to reduce bolierplate code. I use Kotlin for DTOs with Spring 4.1.x (application generated with Jhipster 1.x).

##### **Related issues**

I didn't see any.

##### **Suggest a Fix**

A good option may be to allow the user to choose whether to use plain Java, add lombok annotations, or use the Kotlin language. In the former case, nothing changes, so this feature does not break any existing code. In the case of lombok annotations, the fix should be easy, as it suffices to add the annotations and comment out the setters and getters. In the latter case, no Java DTO class should be generated, in favour of a kt file.

Any comment or suggestion is welcomed!

**Comment by cbornet** [comment]
This has been discussed many times. Check closed issues

**Comment by massimosiani** [comment]
@cbornet I agree about the lombok project, didn't see the closed issues.

However, I do not think that the Kotlin part has ever been discussed. I am not proposing to generate an application in a new server side language, but only to generate a part of the application using Kotlin. There has been much interested around that, and I think it can be useful for some users to have Kotlin preconfigured in their application.

I believe that this deserves at least a discussion.

**Comment by mraible** [comment]
Nothing speaks louder than code! Send us a PR for Kotlin support and we'll consider it. 

However, it's probably best implemented as a module since KHipster doesn't have a great ring to it. 😉

**Comment by cbornet** [comment]
Kotlin has also been discussed. See #2505
Note : I love Kotlin and I think it would make an awesome module but that's a lot of work...

**Comment by PierreBesson** [comment]
@massimosiani Even if you don't want to create the module yourself, maybe you can create a sample project first to show what you want to do.

**Comment by dancancro** [comment]
I'm working on adding a Kotlin feature to my big example app. I got it from one of the [Thinkster Real World backends](https://github.com/gothinkster/kotlin-spring-realworld-example-app). You are all welcome to critique my adaptation of it. The main adjustment I made concerns the User modeling since JHipster precludes adding fields to User. I'm also adding the Angular front end and adapting that one to use ngrx.

https://github.com/dancancro/great-big-example-application/tree/blogwithkotlin

I don't have a problem with KHipster, XHipster, QHipster or any other sort of hipster ;)
